src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -191,16 +204,18 @@ export default class DefaultMessagingSession implements MessagingSession {
     }
   }
 
+  private retryConnection(): boolean {
+    return this.reconnectController.retryWithBackoff(async () => {
+      await this.startConnecting(true);
+    }, null);
+  }
+
   private closeEventHandler(event: CloseEvent): void {
     this.logger.info(`WebSocket close: ${event.code} ${event.reason}`);
-    this.webSocket.destroy();
-    if (
-      !this.isClosing &&
-      this.canReconnect(event.code) &&
-      this.reconnectController.retryWithBackoff(async () => {
-        await this.startConnecting(true);
-      }, null)
-    ) {
+    if (event.code !== 4999) {
+      this.webSocket.destroy();
+    }
+    if (!this.isClosing && this.canReconnect(event.code) && this.retryConnection()) {
```

Comment:
Here, `canReconnect` does not handle `4999` and will return `false`, so, should it now handle this as we are throwing a custom `CloseEvent` with `4999` code and expect a retry and a reconnect?

Reply:
`canReconnect` does handle 4999 cause `closeCode > 4000 ` other than the ones special case returns true.

Reply:
Ah, my bad, missed the `>`, thanks for correcting.